x-pack/plugin/monitoring/src/main/java/org/elasticsearch/xpack/monitoring/collector/node/NodeStatsCollector.java

```diff
@@ -74,8 +74,9 @@ protected boolean shouldCollect(final boolean isElectedMaster) {
             NodesStatsRequest.Metric.PROCESS.metricName(),
             NodesStatsRequest.Metric.THREAD_POOL.metricName(),
             NodesStatsRequest.Metric.FS.metricName());
+        request.timeout(getCollectionTimeout());
 
-        final NodesStatsResponse response = client.admin().cluster().nodesStats(request).actionGet(getCollectionTimeout());
+        final NodesStatsResponse response = client.admin().cluster().nodesStats(request).actionGet();
```

Comment:
Do we need an `ensureNoTimeouts(getCollectionTimeout(), response);` after this line as well?

Reply:
Hmm, I was thinking no since we `throw response.failures().get(0)` anyway, but on reflection that'll be a `FailedNodeException` not the inner timeout. I'll address that.

Reply:
Ok done, see 56ce978ede835827c742b1b5d8a155e21b8f051f.